Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -279,6 +284,60 @@ public async ValueTask<bool> ExistsAsync(Expression<Func<TItem, bool>> predicate
             return count > 0;
         }
 
+        public async ValueTask<IPage<TItem>> ScrollAsync(

```

Comment:
Thinking a bit more about the exposed API, should we call this "PageAsync" instead? The word "Scroll" might seem to imply the intended use case such as a UI. What do you think?

Reply:
I initially went for scroll, due to us aiming to implement paging using offset just wondering if that could be achieved by an overload to `PageAsync` I am easy either way, happy for you to make the call on this one :)

Reply:
Actually, I like this we can do the skip take approach with an API looking like this should fit nicely

```csharp
ValueTask<IPage<TItem>> PageAsync(
            Expression<Func<TItem, bool>> predicate = null,
            int pageNumber = 1,
            int pageSize = 25,
            CancellationToken cancellationToken = default);
```